***Calvin Forinash - Raw Data and Analysis***

In [7]:
import pandas as pd
import numpy as np

data = pd.read_csv("ctf16_crime_data.csv")
data.head()

PK         CCR  HIERARCHY         INCIDENTTIME  \
0  2802309  16000001.0         10  2016-01-01T00:00:00   
1  2803174  16004547.0         11  2016-01-01T00:01:00   
2  2801809  16000367.0          4  2016-01-01T00:10:00   
3  2802315  16000035.0         10  2016-01-01T00:15:00   
4  2802312  16000024.0          4  2016-01-01T00:16:00   

                                INCIDENTLOCATION CLEAREDFLAG  \
0  400 Block North Shore DR Pittsburgh, PA 15212           Y   
1    5400 Block Carnegie ST Pittsburgh, PA 15201           N   
2  500 Block Mt Pleasant RD Pittsburgh, PA 15214           N   
3         300 Block Wood ST Pittsburgh, PA 15222           Y   
4  500 Block Mt Pleasant RD Pittsburgh, PA 15214           N   

          INCIDENTNEIGHBORHOOD INCIDENTZONE  \
0                  North Shore            1   
1          Upper Lawrenceville            2   
2            Northview Heights            1   
3  Golden Triangle/Civic Arena            2   
4            Northview Heights            1   

                      INCIDENTHIERARCHYDESC  \
0            HARRASSMENT/THREAT/ATTEMPT/PHY   
1                        THEFT BY DECEPTION   
2   DISCHARGE OF FIREARM INTO OCC.STRUCTURE   
3            HARRASSMENT/THREAT/ATTEMPT/PHY   
4  PROP MISSILE INTO OCC VEHICLE/OR ROADWAY   

                                            OFFENSES  INCIDENTTRACT  \
0  2702 Aggravated Assault. / 2709(a) Harassment....         2205.0   
1                           3922 Theft by Deception.         1011.0   
2  2707.1 Discharge of a Firearm into Occupied St...         2609.0   
3        2709(a)(3) Harassment No Legitimate Purpose          201.0   
4  2705 Recklessy Endangering Another Person. / 3...         2609.0   

   COUNCIL_DISTRICT  PUBLIC_WORKS_DIVISION          X          Y  
0               1.0                    6.0 -80.012337  40.446263  
1               7.0                    2.0 -79.950295  40.482290  
2               1.0                    1.0 -80.000966  40.478651  
3               6.0                    6.0 -80.001251  40.438918  
4               1.0                    1.0 -80.000966  40.478651

Below, I will analyze this data and put together a list of the top 10 safest neighborhoods in Pittsburgh based on these crime statistics. 

In [8]:
data['INCIDENTNEIGHBORHOOD']

0                         North Shore
1                 Upper Lawrenceville
2                   Northview Heights
3         Golden Triangle/Civic Arena
4                   Northview Heights
                     ...             
239500                       Sheraden
239501                      Brookline
239502                    South Shore
239503              Northview Heights
239504                      Allentown
Name: INCIDENTNEIGHBORHOOD, Length: 239505, dtype: object

In [22]:
sorted_data = data['INCIDENTNEIGHBORHOOD'].value_counts()
sorted_data.head(15)

South Side Flats               13972
Central Business District      12146
Carrick                         8455
Bloomfield                      6563
Shadyside                       6249
East Liberty                    5658
Squirrel Hill South             5436
Homewood South                  5359
Mount Washington                5228
Brookline                       5226
Lincoln-Lemington-Belmar        4831
Knoxville                       4713
Homewood North                  4537
Brighton Heights                4526
Golden Triangle/Civic Arena     4278
Name: INCIDENTNEIGHBORHOOD, dtype: int64

## Least safe neighborhoods ##
Based on this data, the 3 neighborhoods with the most UCR (Uniform Crime Reporting) reports are South Side Flats, the Central Business District and Carrick. 
1. **South Side Flats** - 13972
* This neighborhood sits on the south bank of the Monongahela River, almost directly south of the University of           Pittsburgh. It includes landmarks such as the Color Park and walking trails along the river. The neighborhood is also     centered around E Carson St, which has over 40 bars and pubs; this could be a contributing factor to high crime           reports. 

2. **Central Business District (aka Downtown)** - 12146
* Downtown sits between the Monongahela and Allegheny Rivers. Home to bars, skyscrapers, malls, good food and sports arenas, it is unsurprising that the business center of a historically industrial city is also home to high crime rates. This could also be due simply to the huge amount of traffic, both vehicular and pedestrian, that the neighborhood attracts; more people will eventually lead to more crime. Downtown is often colloquially referred to as The Golden Triangle, which is included separately and has the 15th most UCR reports with 4278. Including this in the Central Business District's tally makes it the neighborhood with the most reports with 16,424, over 3,000 more than any other neighborhood.

3. **Carrick** - 8455
* Everything after South Side Flats and Downtown has significantly fewer crime reports——at least 3700 fewer over the last 5 years. However, Carrick reported almost 2000 more crimes than the next highest, Bloomfield. Carrick sits on the South Side of Pittsburgh. As a densely packed neighborhood in an industrial city, it makes sense that it is a crime hotspot. Based on a quick Google search for "Carrick Pittsburgh" that immediately turned up articles from a few days ago (4/11) about a home invasion in which two men were assaulted. 

However, I want to find the *safest* neighborhoods. There are a handful of neighborhoods with fewer than 600 reports over the last 5+ years (this dataset spans from 2016-present), meaning that they generally get under 100 reports every year. The list below lists the 20 neighborhoods that have officially reported the fewest crimes since 2016. 

In [24]:
reverse_sorted_data = data['INCIDENTNEIGHBORHOOD'].value_counts().iloc[::-1]
reverse_sorted_data.head(20)

Mt. Oliver Boro             72
Mt. Oliver Neighborhood    117
Outside County             154
Outside State              218
Chartiers City             222
New Homestead              248
Ridgemont                  248
Troy Hill-Herrs Island     252
Swisshelm Park             318
East Carnegie              333
Arlington Heights          367
Mount Oliver               377
Summer Hill                394
Hays                       401
Regent Square              408
Oakwood                    439
Esplen                     454
Glen Hazel                 516
Fairywood                  558
St. Clair                  584
Name: INCIDENTNEIGHBORHOOD, dtype: int64

## Safest Neighborhoods by the numbers ##
Based on this data, the 5 safest (actual) neighborhoods are Mount Oliver, Chartiers, New Homestead, Ridgemont and Troy Hill/Herrs Island. I will only go in depth into the top 3 lowest crime neighborhoods. 

1. Mount Oliver - 189
* Mount Oliver is, coincidentally, directly north of Carrick. Although the neighborhood of Mt Oliver is  distinct from Mt Oliver Borough, I am considering them as one neighborhood (that technically isn't part of the city of Pittsburgh). It is a relatively high-end residential neighborhood. According to Wikipedia, the average household size is 2.36, which indicates that its residents are largely couples or single parents, with the occasional large family or person living alone. 

2. Chartiers - 222
* Chartiers is a small neighborhood in the northwest corner of Pittsburgh. With a land area of a mere 84 acres (0.132 sq. miles) and population of less than 500, the crime rate per capita is relatively high. I won't include this neighborhood on the final top 10 list I compile. 

3. New Homestead - 248
* Mostly encompassing wooded areas, New homestead has a population of about 1000 residents in an area of .794 sq miles (again, I won't include this in the final list). 

## Final List - Safest Neighborhoods ##

Below is the final top 10 list of the safest neighborhoods. 
  (*Keep in mind, safest is subjective here, and I included and omitted some neighborhoods based on population or other factors that I felt discounted their statistics*)

#### 1. Mount Oliver ####
* 